src/clients/HubPoolClient.ts

```diff
@@ -646,11 +651,45 @@ export class HubPoolClient {
     this.disputedRootBundles.push(
       ...events["RootBundleDisputed"].map((event) => spreadEventWithBlockNumber(event) as DisputedRootBundle)
     );
-    this.executedRootBundles.push(
-      ...events["RootBundleExecuted"]
-        .filter((event) => !this.configOverride.ignoredHubExecutedBundles.includes(event.blockNumber))
-        .map((event) => spreadEventWithBlockNumber(event) as ExecutedRootBundle)
-    );
+
+    const configStoreVersions: { [blockNumber: number]: number } = {};
+    for (const event of events["RootBundleExecuted"]) {
+      if (this.configOverride.ignoredHubExecutedBundles.includes(event.blockNumber)) {
+        continue;
+      }
+
+      if (!isDefined(configStoreVersions[event.blockNumber])) {
+        const timestamp = (await event.getBlock()).timestamp;
+        const version = this.configStoreClient.getConfigStoreVersionForTimestamp(timestamp);
+        configStoreVersions[event.blockNumber] = version;
+      }
+      const version = configStoreVersions[event.blockNumber];
+
+      const executedRootBundle = spreadEventWithBlockNumber(event) as ExecutedRootBundle;
+      const { l1Tokens, runningBalances } = executedRootBundle;
+      let expectLength: number = l1Tokens.length;
+
+      if (version < UBA_MIN_CONFIG_STORE_VERSION) {
+        // Pre-UBA model: runningBalances length is 1:1 with l1Tokens length.
+        assert([0, 1].includes(version), `Invalid ConfigStore version: ${version}`);
```

Comment:
Nit: prefer reusing some of the config store client logic to validate the version. Like isValidVersion

Reply:
I ended up removing the assert since it was mostly useful during development and test.